internal/provider/fake_control_api_test.go

```
@@ -0,0 +1,598 @@
+// ~/.terraformrc dev_overrides would otherwise run a stale installed binary
+// (the trap that the Phase 0c spike and CODEGEN_STRATEGY.md call out). Building
+// here guarantees the tests exercise THIS code with no network install.
+//
+// It returns the temp directory it created so the caller can remove it.
+func setupHermeticProvider() (string, error) {
+	dir, err := os.MkdirTemp("", "ably-hermetic")
+	if err != nil {
+		return "", err
+	}
+	bin := filepath.Join(dir, "terraform-provider-ably")
+	build := exec.Command("go", "build", "-o", bin, "github.com/ably/terraform-provider-ably")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)